### PR TITLE
Patch round 2024-05

### PR DESCRIPTION
**Pull request by matijs**
- husky 9 configuration
  - Clean up hook scripts
  - Use pnpm instead of npx in hook scripts
  - Replace package.json#postinstall with package.json#prepare

- install patch updates
- install minor updates
- install major updates, prevent updating the following deps:
  - eslint, eslint-plugin-json because the ecosystem is not ready yet
  - storybook, storybook/* because it is too much effort right now
  - prettier because prettier 3 is async

- Rename to .cjs for clarity
- Make sure that lint-staged picks up .mjs and .cjs files
- Make sure that eslint picks up .cjs files

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,4 +1 @@
-#!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
-
-npx --no-install commitlint --edit "$1"
+pnpm commitlint --edit "$1"
```
Comment (Robbert) [inline .husky/commit-msg:1]: moet dit `pnpm exec` zijn?

File: .husky/pre-commit [hunk 2]
```diff
@@ -1,3 +1 @@
-#!/bin/sh
-
-npx lint-staged
\ No newline at end of file
+pnpm lint-staged
```
Comment (Robbert) [inline .husky/pre-commit:1]: moet dit `pnpm exec` zijn?

Comment (Robbert) [inline .husky/pre-commit:1]:
```suggestion
pnpm exec lint-staged
```

File: .husky/commit-msg (hunk 1 above)
Comment (Robbert) [inline .husky/commit-msg:1]:
```suggestion
pnpm exec commitlint --edit "$1"
```

**Comment by Robbert** [comment]
"47 tests fail to render"

@matijs er gaat nog iets mis: https://62d1558ac36c853a5aaad43f-qhvffahbuo.chromatic.com/?path=/story/templates-login--login

Werkt wel in `main`: https://nl-design-system.github.io/denhaag/?path=/story/templates-login--login

**Review by Robbert — CHANGES_REQUESTED** [review]

**Review by Borai — COMMENTED** [review]
LGTM 🚀

**Comment by matijs** [comment]
Closing in favour of #1537

**Pull request closed without merging**